sys/net/gnrc/link_layer/gomach/gomach.c

```diff
@@ -0,0 +1,2222 @@
+    gnrc_gomach_set_timeout(netif, GNRC_GOMACH_TIMEOUT_CP_MAX, GNRC_GOMACH_CP_DURATION_MAX_US);
+
+    gnrc_netif_set_rx_started(netif, false);
+    gnrc_gomach_set_pkt_received(netif, false);
+    netif->mac.gomach.cp_extend_count = 0;
+    gnrc_gomach_set_quit_cycle(netif, false);
+    gnrc_gomach_set_unintd_preamble(netif, false);
+    gnrc_gomach_set_beacon_fail(netif, false);
+    gnrc_gomach_set_cp_end(netif, false);
+    gnrc_gomach_set_got_preamble(netif, false);
+
+    /* Flush RX queue and turn on radio. */
+    gnrc_priority_pktqueue_flush(&netif->mac.rx.queue);
+    gnrc_gomach_set_netdev_state(netif, NETOPT_STATE_IDLE);
+
+    /* Turn to current public channel. */
+    gnrc_gomach_turn_channel(netif, netif->mac.gomach.cur_pub_channel);
+
+    /* Enable Auto-ACK for data packet reception. */
+    gnrc_gomach_set_autoack(netif, NETOPT_ENABLE);
+
+    netif->mac.rx.listen_state = GNRC_GOMACH_LISTEN_CP_LISTEN;
+    gnrc_gomach_set_update(netif, false);
+}
+
+static void gomach_listen_cp_listen(gnrc_netif_t *netif)
+{
+    if (gnrc_gomach_get_pkt_received(netif)) {
+        gnrc_gomach_set_pkt_received(netif, false);
+        gnrc_gomach_cp_packet_process(netif);
+
+        /* If the device has replied a preamble-ACK, it must waits for the data.
+         * Here, we extend the CP. */
+        if (gnrc_gomach_get_got_preamble(netif)) {
+            gnrc_gomach_set_got_preamble(netif, false);
+            gnrc_gomach_set_cp_end(netif, false);
+            gnrc_gomach_clear_timeout(netif, GNRC_GOMACH_TIMEOUT_CP_END);
+            gnrc_gomach_set_timeout(netif, GNRC_GOMACH_TIMEOUT_CP_END, GNRC_GOMACH_CP_DURATION_US);
+        }
+        else if ((!gnrc_gomach_get_unintd_preamble(netif)) &&
+                 (!gnrc_gomach_get_quit_cycle(netif))) {
+            gnrc_gomach_set_got_preamble(netif, false);
+            gnrc_gomach_set_cp_end(netif, false);
+            gnrc_gomach_clear_timeout(netif, GNRC_GOMACH_TIMEOUT_CP_END);
+            gnrc_gomach_set_timeout(netif, GNRC_GOMACH_TIMEOUT_CP_END, GNRC_GOMACH_CP_DURATION_US);
+        }
+    }
+
+    /* If we have reached the maximum CP duration, quit CP. */
+    if (gnrc_gomach_timeout_is_expired(netif, GNRC_GOMACH_TIMEOUT_CP_MAX)) {
+        gnrc_gomach_clear_timeout(netif, GNRC_GOMACH_TIMEOUT_WAIT_RX_END);
+        gnrc_gomach_clear_timeout(netif, GNRC_GOMACH_TIMEOUT_CP_END);
+        gnrc_gomach_clear_timeout(netif, GNRC_GOMACH_TIMEOUT_CP_MAX);
+        netif->mac.rx.listen_state = GNRC_GOMACH_LISTEN_CP_END;
+        gnrc_gomach_set_update(netif, true);
+        return;
+    }
+
+    if ((gnrc_gomach_timeout_is_expired(netif, GNRC_GOMACH_TIMEOUT_CP_END))) {
+        gnrc_gomach_set_cp_end(netif, true);
+        gnrc_gomach_clear_timeout(netif, GNRC_GOMACH_TIMEOUT_CP_END);
+    }
+
+    /* If CP duration timeouted or we must quit CP, go to CP end. */
+    if (gnrc_gomach_get_cp_end(netif) || gnrc_gomach_get_quit_cycle(netif)) {
+  
```

Comment:
Same goes for this switch-case.

Reply:
Em.. this is not a switch-case?

Reply:
It used to be, didn't it? I can't even find this comment in the current patch oO

Reply:
No, still, I didn't ever put a switch here, the two get functiona only gets the state of related flags. Anyway, will tackle the large code block issue in swicth. :-)

Reply:
Thanks. As I said. Don't know where the line went, I was talking about :-/